### ISSUE TITLE
False positive in ProtectedMemberInFinalClass with JVM finalizer

### ISSUE THREAD

**Issue by ccjernigan** [detekt/detekt#5660]
## Expected Behavior
JVM finalizer methods shouldn't trigger detekt warnings

## Observed Behavior
`Member with protected visibility in final class is private. Consider using private or internal as modifier. [ProtectedMemberInFinalClass]`

## Steps to Reproduce

Create a class for the JVM that implements the finalize API.  See 

E.g.
```
class MyFinalizable {
    @Throws(Throwable::class)
    protected fun finalize() {
    }
}
```

## Context

https://kotlinlang.org/docs/java-interop.html#finalize

## Your Environment

* Version of detekt used: 1.21.0
* Version of Gradle used (if applicable):
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version:
* Link to your project (if it's a public repository):
